[Bug]: after update runtime-corejs to 7.19.1 webpack falling with error

### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

babel-loader (webpack)

### Input code

my package json babel's part:

```
"devDependencies": {
    "@babel/core": "^7.18.6",
    "@babel/preset-env": "^7.18.6",
    "@babel/preset-flow": "^7.18.6",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "another_dependency": "some_version"
}
```

one of my dependency needs "@babel/runtime-corejs2": "^7.18.6"


### Configuration file name

_No response_

### Configuration

_No response_

### Current and expected behavior

After update babel/runtime-corejs2 to 7.19.1 my webpack build ends with errors:

```
ERROR in ./node_modules/@babel/runtime-corejs2/core-js/date/now.js 1:0-55
Module not found: Error: Can't resolve 'core-js/library/fn/date/now' in 'path/node_modules/@babel/runtime-corejs2/core-js/date'
resolve 'core-js/library/fn/date/now' in 'path/node_modules/@babel/runtime-corejs2/core-js/date'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/date)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/date/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/date/now)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/date/now doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/date/now.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/date/now.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/date/now.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/date/now doesn't exist
    
ERROR in ./node_modules/@babel/runtime-corejs2/core-js/number/is-integer.js 1:0-64
Module not found: Error: Can't resolve 'core-js/library/fn/number/is-integer' in 'path/node_modules/@babel/runtime-corejs2/core-js/number'
resolve 'core-js/library/fn/number/is-integer' in 'path/node_modules/@babel/runtime-corejs2/core-js/number'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/number)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/number/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/number/is-integer)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/number/is-integer doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/number/is-integer.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/number/is-integer.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/number/is-integer.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/number/is-integer doesn't exist

ERROR in ./node_modules/@babel/runtime-corejs2/core-js/object/assign.js 1:0-60
Module not found: Error: Can't resolve 'core-js/library/fn/object/assign' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
resolve 'core-js/library/fn/object/assign' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/object)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/object/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/object/assign)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/assign doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/assign.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/assign.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/assign.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/object/assign doesn't exist

ERROR in ./node_modules/@babel/runtime-corejs2/core-js/object/create.js 1:0-60
Module not found: Error: Can't resolve 'core-js/library/fn/object/create' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
resolve 'core-js/library/fn/object/create' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/object)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/object/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/object/create)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/create doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/create.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/create.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/create.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/object/create doesn't exist

ERROR in ./node_modules/@babel/runtime-corejs2/core-js/object/keys.js 1:0-58
Module not found: Error: Can't resolve 'core-js/library/fn/object/keys' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
resolve 'core-js/library/fn/object/keys' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/object)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/object/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/object/keys)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/keys doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/keys.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/keys.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/keys.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/object/keys doesn't exist

ERROR in ./node_modules/@babel/runtime-corejs2/core-js/object/set-prototype-of.js 1:0-70
Module not found: Error: Can't resolve 'core-js/library/fn/object/set-prototype-of' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
resolve 'core-js/library/fn/object/set-prototype-of' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/object)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/object/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/object/set-prototype-of)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/set-prototype-of doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/set-prototype-of.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/set-prototype-of.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/set-prototype-of.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/object/set-prototype-of doesn't exist

ERROR in ./node_modules/@babel/runtime-corejs2/core-js/object/values.js 1:0-60
Module not found: Error: Can't resolve 'core-js/library/fn/object/values' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
resolve 'core-js/library/fn/object/values' in 'path/node_modules/@babel/runtime-corejs2/core-js/object'
  Parsed request is a module
  using description file: path/node_modules/@babel/runtime-corejs2/package.json (relative path: ./core-js/object)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      path/node_modules/@babel/runtime-corejs2/core-js/object/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/core-js/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/runtime-corejs2/node_modules doesn't exist or is not a directory
      path/node_modules/@babel/node_modules doesn't exist or is not a directory
      path/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in path/node_modules
        existing directory path/node_modules/core-js
          using description file: path/node_modules/core-js/package.json (relative path: .)
            using description file: path/node_modules/core-js/package.json (relative path: ./library/fn/object/values)
              no extension
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/values doesn't exist
              .js
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/values.js doesn't exist
              .json
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/values.json doesn't exist
              .wasm
                Field 'browser' doesn't contain a valid alias configuration
                path/node_modules/core-js/library/fn/object/values.wasm doesn't exist
              as directory
                path/node_modules/core-js/library/fn/object/values doesn't exist

webpack 5.74.0 compiled with 7 errors in 8770 ms
```

### Environment

- Babel/runtime-corejs2: 7.19.1
- Babel-loader version: 8.2.5
- Node: 18.9.0
- npm: 8.19.2
- OS: macOS 12.2.1

### Possible solution

downgrade to 7.19.0

### Additional context

_No response_

--- a/constraints.pro
+++ b/constraints.pro
@@ -92,7 +92,7 @@ gen_enforced_field(WorkspaceCwd, 'exports', '{ ".": "./lib/index.js", "./package
 gen_enforced_field(WorkspaceCwd, 'type', 'commonjs') :-
   \+ workspace_field(WorkspaceCwd, 'type', 'module').
 
-% Enforce a default 'conditions', unless it's already specified
+% Enforces a default 'conditions', unless it's already specified
 gen_enforced_field(WorkspaceCwd, 'conditions', '{ "USE_ESM": [{ "type": "module" }, null] }') :-
   \+ workspace_field(WorkspaceCwd, 'private', true),
   \+ workspace_field(WorkspaceCwd, 'conditions', _),
@@ -100,3 +100,12 @@ gen_enforced_field(WorkspaceCwd, 'conditions', '{ "USE_ESM": [{ "type": "module"
   % Exclude some packages
   workspace_ident(WorkspaceCwd, WorkspaceIdent),
   WorkspaceIdent \= '@babel/compat-data'.
+
+% Enforces that @babel/runtime-corejs2 must depend on core-js 2
+gen_enforced_dependency(WorkspaceCwd, 'core-js', '^2.6.12', DependencyType) :-
+  % Get the workspace name
+  workspace_ident(WorkspaceCwd, WorkspaceIdent),
+  % Only consider 'dependencies'
+  (DependencyType = 'dependencies'),
+  % The rule works for @babel/runtime-corejs2 only
+  (WorkspaceIdent = '@babel/runtime-corejs2').

--- a/packages/babel-runtime-corejs2/package.json
+++ b/packages/babel-runtime-corejs2/package.json
@@ -14,7 +14,7 @@
   "homepage": "https://babel.dev/docs/en/next/babel-runtime-corejs2",
   "author": "The Babel Team (https://babel.dev/team)",
   "dependencies": {
-    "core-js": "^3.25.1",
+    "core-js": "^2.6.12",
     "regenerator-runtime": "^0.13.4"
   },
   "exports": {

